### DISCUSSION TITLE
Local file changes detected, aborting deployment

### DISCUSSION THREAD

**Discussion by ssewalt** [dsmrreader/dsmr-reader#1491] [Questions & answers / How-to's] [answered]
ik probeer te updaten van 4.16.3 naar 4.17.1.
Gewoon via ./deploy.sh

Wat ik ook probeer ik blijf melding krijgen 

pi@DSMR:~ $ sudo su - dsmr
(dsmrreader) dsmr@DSMR:~/dsmr-reader $ ./deploy.sh

 --- You are currently running version:
4.16.3

 --- Checking Python version.
Running Python 3.7.3 (default, Jan 22 2021, 20:04:44)
[GCC 8.3.0]

 --- Checking for local file changes.
 M backups/.gitkeep
 M backups/archive/.gitkeep
 M backups/archive/README.txt

[!] Local file changes detected, aborting deployment
    To revert all files to their default, WITHOUT backup, execute:

    git reset --hard HEAD

Geen idee hoe op te lossen.

**Comment by dennissiemensma** [comment]
Bedankt voor je melding. Je kunt het commando onderaan de melding uitvoeren en daarna nogmaals een deploy proberen.

**Comment by ssewalt** [comment]
Dennis, dank voor je terugkoppeling.  De git-reset uitgevoerd en daarna nogmaals de deploy geprobeerd; maar helaas blijf dezelfde melding krijgen. Dit vaker uitgevoerd; maar het nummer van de HEAD blijft gelijk.  Wat ik verder ook geprobeerd heb is dsmr reader uitgezet en dan de deploy. maar melding blijft gelijk.

**Comment by dennissiemensma** [comment]
Wat is de output van `git reset --hard HEAD`?

Je kunt anders nog proberen de bestanden te verwijderen en daarna de reset te doen, zodat ze weer teruggezet worden. Het zijn allemaal placeholders of onbelangrijke bestanden voor de inhoudelijke werking.

```
sudo su - dsmr
rm backups/.gitkeep backups/archive/.gitkeep backups/archive/README.txt
git reset --hard HEAD
```

**Comment by dennissiemensma** [comment]
Ik heb dit issue even verplaatst naar een nieuw onderdeel binnen Github, waar ik alle issues met vraag/antwoord/hulp naar verplaatst heb net: https://github.com/dsmrreader/dsmr-reader/discussions/1491

**Comment by ssewalt** [comment]
Dennis, helaas heeft het verwijderen van die bestanden ook niet geholpen. 
Hier ook output van de git reset:
```
(dsmrreader) dsmr@DSMR:~/dsmr-reader $ ./deploy.sh


 --- You are currently running version:
4.16.3


 --- Checking Python version.
Running Python 3.7.3 (default, Jan 22 2021, 20:04:44)
[GCC 8.3.0]


 --- Checking for local file changes.
 D backups/.gitkeep
 D backups/archive/.gitkeep
 D backups/archive/README.txt

[!] Local file changes detected, aborting deployment
    To revert all files to their default, WITHOUT backup, execute:

    git reset --hard HEAD

(dsmrreader) dsmr@DSMR:~/dsmr-reader $ git reset --hard HEAD

HEAD is now at 5a845e29 Rework issue templates

(dsmrreader) dsmr@DSMR:~/dsmr-reader $
```

**Comment by dennissiemensma** [comment]
Wat is de output van:

```
sudo su - dsmr

git status
git branch
```

**Comment by ssewalt** [comment]
bij deze:

```
pi@DSMR:~ $ sudo su - dsmr
(dsmrreader) dsmr@DSMR:~/dsmr-reader $ git status
On branch v4
Your branch is up to date with 'origin/v4'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   backups/.gitkeep
        modified:   backups/archive/.gitkeep
        modified:   backups/archive/README.txt

no changes added to commit (use "git add" and/or "git commit -a")
(dsmrreader) dsmr@DSMR:~/dsmr-reader $ git branch
* v4
(dsmrreader) dsmr@DSMR:~/dsmr-reader $
```

**Comment by dennissiemensma** [comment]
Wat is de output van:
 
```
sudo su - dsmr

git diff
ls -l backups/.gitkeep backups/archive/.gitkeep backups/archive/README.txt
```

Het is namelijk vast iets subtiels wat blokkeert, zoals bestandsrechten of iets dergelijks.

**Comment by ssewalt** [comment]
ik verwacht ook dat het iets subtiels is; alleen mijn kennis in net niet op orde om het zelf op te lossen. Ben dus zeer blij met deze mogelijkheid.... Hierbij de output:
```
(dsmrreader) dsmr@DSMR:~/dsmr-reader $ git diff
diff --git a/backups/.gitkeep b/backups/.gitkeep
old mode 100644
new mode 100755
diff --git a/backups/archive/.gitkeep b/backups/archive/.gitkeep
old mode 100644
new mode 100755
diff --git a/backups/archive/README.txt b/backups/archive/README.txt
old mode 100644
new mode 100755

(dsmrreader) dsmr@DSMR:~/dsmr-reader $ ls -l backups/.gitkeep backups/archive/.gitkeep backups/archive/README.txt

-rwxr-xr-x 1 dsmr dsmr   0 okt  1 19:36 backups/archive/.gitkeep
-rwxr-xr-x 1 dsmr dsmr 109 okt  1 19:36 backups/archive/README.txt
-rwxr-xr-x 1 dsmr dsmr   0 okt  1 19:36 backups/.gitkeep
(dsmrreader) dsmr@DSMR:~/dsmr-reader $
```

**Comment by dennissiemensma** [comment]
Helder, het zijn inderdaad de permissies! Je kunt dan dit proberen:

```
sudo su - dsmr

chmod 0644 backups/.gitkeep backups/archive/.gitkeep backups/archive/README.txt

git diff
```

Als het goed is geeft `git diff` dan geen wijzigingen meer.

Daarna:
```
./deploy.sh
```

**Comment by ssewalt** [comment]
er is uiteindelijk niets veranderd.... ik vroeg mij af wat zouden de rechten moeten zijn op de backups/.gitkeep backups/archive/.gitkeep backups/archive/README.txt files?

**Comment by dennissiemensma** [comment]
Verandert het chmod commando niet de rechten? In de git diff kun je zien wat de standaard zou moeten zijn, 0644.

**Comment by ssewalt** [comment]
Inderdaad de rechten veranderen niet. Welke ik ook probeer. Er gebeurt niets qua rechten verandering

⁣

**Comment by dennissiemensma** [comment]
Wat geeft:

```
ls -l /home/dsmr/dsmr-reader/backups/
```

En werkt dit ook niet?

```
sudo chmod 0644 /home/dsmr/dsmr-reader/backups/.gitkeep /home/dsmr/dsmr-reader/backups/archive/.gitkeep /home/dsmr/dsmr-reader/backups/archive/README.txt
```

Als ik het hier probeer herstelt ofwel een `git reset --hard HEAD` ofwel de `chmod 0644 ...` de rechten, dus ik snap niet zo goed wat het dan blokkeert. Ook niet omdat de eigenaar van de bestanden ook goed lijkt te staan op `dsmr`.

**Comment by ssewalt** [comment]
dennis; wat ik ook probeer ook via de sudo chmod er veranderd niets qua rechten op de backups....Ze staan nu zo...
drwxr-xr-x 2 dsmr dsmr       0 okt  3 20:30 archive
-rwxr-xr-x 1 dsmr dsmr 3568531 okt  2 00:30 dsmrreader-postgresql-backup-Friday.sql.gz
-rwxr-xr-x 1 dsmr dsmr 3538468 sep 28 00:30 dsmrreader-postgresql-backup-Monday.sql.gz
-rwxr-xr-x 1 dsmr dsmr 3580528 okt  3 00:29 dsmrreader-postgresql-backup-Saturday.sql.gz
-rwxr-xr-x 1 dsmr dsmr 3586763 okt  4 00:30 dsmrreader-postgresql-backup-Sunday.sql.gz
-rwxr-xr-x 1 dsmr dsmr 3555688 okt  1 00:30 dsmrreader-postgresql-backup-Thursday.sql.gz
-rwxr-xr-x 1 dsmr dsmr 3544046 sep 29 00:30 dsmrreader-postgresql-backup-Tuesday.sql.gz
-rwxr-xr-x 1 dsmr dsmr 3554104 sep 30 00:30 dsmrreader-postgresql-backup-Wednesday.sql.gz
pi@DSMR:~ $
En dat blijven ze ook.

**Comment by dennissiemensma** [comment]
Kan het zijn dat je een mount of iets gebruikt? 

Ik heb verder geen directe oplossingen meer voor je. Op termijn kan ik de bestanden uit versiebeheer halen en moet ik ze vervangen met een script wat ervoor zorgt dat die mappen na installatie bestaan (daarvoor zijn die bestanden).

Het laatste wat je kunt doen is lokaal de volgende regels uitcommenten met een `#` voor, in `deploy.sh`: https://github.com/dsmrreader/dsmr-reader/blob/v4/deploy.sh#L27-L41
Daarmee zet je de check uit en daarna is het de vraag of het in 1x goed gaat. Zorg er wel voor dat je een database backup hebt, al verwacht ik geen issues in die specifieke bestanden.

**Comment by ssewalt** [comment]
Dsmr draait op een aparte rpi bij mij volledig standalone en via mqtt naar home assistent.. geen mount oid aanwezig 
Ik ga morgen jouw aangepaste deploy proberen. Wordt vervolgd.

⁣Verzonden door Blue ​

Op 4 okt. 2021 21:48, om 21:48, Dennis Siemensma ***@***.***> schreef:

**Reply by ssewalt** [reply]
Helaas heeft dat ook niet geholpen uiteindelijk; systeem meldt dat hij gewoon de versie 4.16.3 geinstalleerd heeft...
 --- Reloading running apps...

 * Reloading process: dsmr_webinterface (Gunicorn)   [OK]
 * Reloading process: dsmr_backend                   [OK]
 * Reloading process: dsmr_datalogger                [OK]


 --- Clearing cache...


 --- Deployed version:
4.16.3

Vrees dat er niets anders op zit dan een nieuwe install te doen. Kan ik daarna de backup terug zetten en dan ben ik weer terug ?

**Comment by ssewalt** [comment, marked as answer]
fixed....het bleek dat ik inderdaad toch een mount had gemaakt naar de backup dir... die tijdelijk verwijderd en nu ging het goed.

**Comment by dennissiemensma** [comment]
Fijn om te horen, dank voor je aanvulling!
